airbyte-config/config-persistence/src/main/java/io/airbyte/config/persistence/split_secrets/SecretPersistence.java

```diff
@@ -68,4 +68,12 @@ static Optional<SecretPersistence> getEphemeral(final DSLContext dslContext, fin
     }
   }
 
+  static SecretsHydrator getDataPlaneSecretsHydrator(final Configs configs) {
```

Comment:
@cgardens discussion from parent PR:
![image](https://user-images.githubusercontent.com/7798112/186285708-647bf879-266b-458d-95e0-d21e83065643.png)

Reply:
If we do keep it as 2 methods then i think this one should be called `getGsmSecretsHydrator`.

Reply:
Also, part of me is leaning toward a `SecretsHydratorFactory.get` that would return the expected implementation given the environment ({dev,prod}, {control_plane,data_plane}) rather than bundling it in `SecretPersistence`.

This should be more aligned with the dependency injection pattern that should with micronaut.

Reply:
+1

Reply:
I had already started on a refactor along the lines of what Charles initially suggested with the single method. In the interest of landing this PR and unblocking Micronaut, I'll leave things as they are, since it keeps changes to a minimum. I'll add a TODO to migrate to a factory pattern that works well with Micronaut for the near future. Davin and I will be working closely with Jonathan on conversion of the WorkerApp to Micronaut so this can be something we address at that point. Hopefully that sounds reasonable to the both of you, but please let me know if you feel strongly another way!

Reply:
Works for me, as long as we commit to it, unblocking both work streams is probably more pressing than refactoring this factory.